src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+            {
+                var queryClauseInfo = _semanticModel.GetQueryClauseInfo(joinClause, _cancellationToken);
+                if (queryClauseInfo.OperationInfo.Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.Parameters.Length == 4)
+                {
+                    processingState.AddVariableName(joinClause.Identifier.ValueText);
+                    if (joinClause.Into != null)
+                    {
+                        var intoDeclaration =
+                            SyntaxFactory.LocalDeclarationStatement(
+                                SyntaxFactory.VariableDeclaration(
+                                    VarNameIdentifier,
+                                    SyntaxFactory.SingletonSeparatedList(
+                                        SyntaxFactory.VariableDeclarator(
+                                            joinClause.Into.Identifier,
+                                            argumentList: null,
+                                            SyntaxFactory.EqualsValueClause(
+                                                SyntaxFactory.AnonymousObjectCreationExpression(
+                                                    SyntaxFactory.SeparatedList(
+                                                        processingState.VariableNames.Select(vn => CreateAnonymousObjectMemberDeclarator(vn)))))))));
+                        processingState.AddStatementProcessingMethod(s => CreateJoinStatement(joinClause, AddToBlockTop(intoDeclaration, s)));
+                    }
+                    else
+                    {
+                        processingState.AddStatementProcessingMethod(s => CreateJoinStatement(joinClause, s));
+                    }
+
+                    return true;
+                }
+
+                return false;
+            }
+
+            private static BlockSyntax AddToBlockTop(StatementSyntax newStatement, StatementSyntax statement)
+            {
+                if (statement is BlockSyntax block)
+                {
+                    return block.WithStatements(block.Statements.Insert(0, newStatement));
+                }
+                else
+                {
+                    return SyntaxFactory.Block(newStatement, statement);
+                }
+            }
+
+            private static BlockSyntax WrapWithBlock(StatementSyntax statement)
+                => statement is BlockSyntax block ? block : SyntaxFactory.Block(statement);
+
+            private static ExpressionSyntax ProcessCondition(ExpressionSyntax expression)
+            {
+                if (expression.Kind() == SyntaxKind.ParenthesizedExpression)
+                {
+                    expression = ((ParenthesizedExpressionSyntax)expression).Expression;
+                }
+
+                return expression.WithoutTrivia();
+            }
+
+            private static AnonymousObjectMemberDeclaratorSyntax CreateAnonymousObjectMemberDeclarator(string name)
```

Comment:
>AnonymousObjectMemberDeclaratorSyntax [](start = 27, length = 37)

It doesn't look like this function is used anywhere.

Reply:
used for joinClause.Into. moved it there as a local funciton

---
In reply to: [177205131](https://github.com/dotnet/roslyn/pull/25362#discussion_r177205131) [](ancestors = 177205131)